### PR TITLE
Adding container maptrix and SpMV.

### PR DESCRIPTION
**Pull request by ancysarahtom**

### CODE REVIEW

File: examples/reddit_pagerank.cpp [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+  int N_old = pr_old.size();
+  int N_new = pr_new.size();
+  //std::cout << "Prev iter size: " << N_old << ", Current iter size: " << N_new << std::endl;
+
+  auto pr_old_ptr = pr_old.get_ygm_ptr();
+  auto pr_new_ptr = pr_new.get_ygm_ptr();
+
+  map_type norm_map(world);
+  auto norm_map_ptr = norm_map.get_ygm_ptr();
```
Comment (steiltre) [inline examples/reddit_pagerank.cpp:34]: You can do this without a norm_map that gathers residuals component by component. Instead, define a `static double` that holds the piece of the squared norm local to the rank.

A static variable can be used inside of a lambda (even one spawned by a visit) without being explicitly passed (either as an argument or by capturing).

Replace the definition of norm_map with
`static double local_norm_squared;
local_norm_squared = 0.0;`

File: examples/reddit_pagerank.cpp [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    std::cout << argv[i];
+  }
+
+  // Building the author-author graph.
+  ygm::io::ndjson_parser json_parser(world, fnames);
+  json_parser.for_all([&A, &deg, &deg_acc_lambda, &A_acc_lambda](auto &json_line) {
+    if (json_line["LL_author"] != "AutoModerator" &&
+       json_line["LL_parent_author"] != "AutoModerator") {
+
+      /* Reading it in backward.. */
```
Comment (steiltre) [inline examples/reddit_pagerank.cpp:123]: Comment is no longer accurate...

File: include/ygm/container/experimental/detail/maptrix_impl.hpp [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    m_csr.swap(s.csr);
+    m_csc.swap(s.csc);
+  }
+
+ protected:
+  maptrix_impl() = delete;
+
+  value_type                                          m_default_value;
+  csr_impl                                            m_csr;      
+  csc_impl                                            m_csc;
```
Comment (steiltre) [inline include/ygm/container/experimental/detail/maptrix_impl.hpp:139]: Change the 'csr' and 'csc' naming to 'row_view' and 'column_view'

**Comment by ancysarahtom** [comment]
>

File: include/ygm/container/experimental/detail/maptrix_impl.hpp (hunk 3 above)
Comment (ancysarahtom) [reply include/ygm/container/experimental/detail/maptrix_impl.hpp:139]: Yep, will do.

File: examples/reddit_pagerank.cpp [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+      auto dst = json_line["LL_author"].as_string(); 
+      auto src = json_line["LL_parent_author"].as_string();
```
Comment (steiltre) [inline examples/reddit_pagerank.cpp:125]: This creates a copy of src and dst in each line. Declare these as `const auto &`

File: examples/reddit_pagerank.cpp (hunk 1 above)
Comment (ancysarahtom) [reply examples/reddit_pagerank.cpp:34]: Changed this, thanks!

**Pull request merged**